recipes/configs/llama3_1/kd_single_device.yaml

```diff
@@ -0,0 +1,110 @@
+# Config for single device knowledge distillation in kd_single_device.py
+# using a teacher and student model
+#
+# This config assumes that you've run the following commands before launching
+# this run:
+#   tune download meta-llama/Meta-Llama-3.1-8B-Instruct --output-dir /tmp/Meta-Llama-3.1-8B-Instruct
+#   tune run lora_finetune_single_device --config llama3_1/8B_lora_single_device
+#
+# To launch on a single device, run the following command from root:
+#   tune run kd_single_device --config llama3_1/kd_single_device
+#
+# This config works only for distilling on a single device.
+
+
+# Model Arguments
+model:
+  _component_: torchtune.models.llama3_1.lora_llama3_1
+  lora_attn_modules: ['q_proj', 'v_proj']
+  apply_lora_to_mlp: False
+  apply_lora_to_output: False
+  vocab_size: 128256
+  num_layers: 16
+  num_heads: 32
+  num_kv_heads: 8
+  embed_dim: 2048
+  max_seq_len: 131072
+  intermediate_dim: 8192
+  attn_dropout: 0.0
+  norm_eps: 1e-5
+  rope_base: 500000.0
+  lora_rank: 8
+  lora_alpha: 16
+  lora_dropout: 0.05
+  use_dora: False
+  quantize_base: False
+
+teacher_model:
+  _component_: torchtune.models.llama3_1.llama3_1_8b
+
+# Tokenizer
+tokenizer:
+  _component_: torchtune.models.llama3.llama3_tokenizer
+  path: /tmp/Meta-Llama-3.1-8B-Instruct/original/tokenizer.model
+  max_seq_len: null
+
+checkpointer:
+  _component_: torchtune.training.FullModelMetaCheckpointer
+  checkpoint_dir: /tmp/Llama-3.1-Student/
+  checkpoint_files: [
+    consolidated.00.pth
+  ]
+  recipe_checkpoint: null
+  output_dir: /tmp/kd_model/
+  model_type: LLAMA3
+
+# Teacher checkpoint
+teacher_checkpointer:
+  _component_: torchtune.training.FullModelMetaCheckpointer
+  checkpoint_dir: /tmp/Meta-Llama-3.1-8B-Instruct/lora_finetuned_single_device_epoch_1/
```

Comment:
Should make sure that this directory matches the output of whatever command you give for LoRA finetuning at the top of this file